src/main/java/hudson/plugins/plot/Plot.java

```diff
@@ -455,7 +465,7 @@ private int getHeight() {
         return height;
     }
 
-    public AbstractProject<?, ?> getProject() {
+    public Job<?, ?> getProject() {
```

Comment:
Are there any other plugins depending on this plugin? If so, this is a risk of breaking compatibility by returning non-AbstractProject types of Jobs

Reply:
I've searched for [usage in github](https://github.com/search?l=XML&p=1&q=org%3Ajenkinsci+plot&type=Code) and found that only PHP plugin [depends on plot plugin version 1.5](https://github.com/jenkinsci/php-plugin/blob/master/pom.xml#L64). 
What's the best way to do it in this case?
[docs](https://github.com/jenkinsci/pipeline-plugin/blob/master/DEVGUIDE.md) tell to replace `AbstractProject` with `Job` if I'm getting that correctly:
> For Pipeline compatibility, plugins formerly referring to AbstractProject/AbstractBuild will generally need to start dealing with Job/Run

Reply:
:+1:  That's the right way to search for depending plugins.  Unfortunately although PHP plugin hasn't had a release in years, it's got 1000 active installs apparently, so we do need to worry about compatibility a bit. 

> or what means "start dealing" in this case if not "replace"? support both versions with overloaded methods?

It means that someone didn't do a good job communicating when writing that doc.  :boom: 

So, in plugins with no downstream plugins dependent on public/protected fields and methods with AbstractProject/AbstractBuild, you simply replace with Job/Run in most cases.

With dependent plugins you generally change the methods in place to accept a Job/Run as appropriate, but create a shim method that takes the old parameters and invokes the new method (so, usually just 3 lines).

I know, I know, it feels messy but it's the easiest way to maintain binary compatibility.